### PR TITLE
Server database - First draft

### PR DESCRIPTION
**Pull request by ZQUEMA**
Add a first draft of the database implementation

### CODE REVIEW

File: libs/database/database.h [hunk 1]
```diff
@@ -7,14 +7,66 @@
...
+
+/**
+ * @brief It searches the database for a user with the given id
+ *
+ * @param db The database to search in.
+ * @param id The id of the user to search for.
+ *
+ * @return A pointer to a user_t struct.
+ */
+user_t *db_search_user_by_id(database_t *db, const id_t id)
```
Comment (aurelenc) [inline libs/database/database.h:62]: missing semicolon ?

File: libs/database/database.h [hunk 2]
```diff
@@ -7,14 +7,66 @@
...
+
+/**
+ * @brief It searches the database for a user with the given pseudo
+ *
+ * @param db the database
+ * @param pseudo the pseudo of the user to search for
+ *
+ * @return A pointer to the user_t struct.
+ */
+user_t *db_search_user_by_pseudo(database_t *db, const char *pseudo)
```
Comment (aurelenc) [inline libs/database/database.h]: same

File: tests/tests_llist_search.c [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+** Teams
+** File description:
+** tests_llist_search
+*/
+
+#include "llist.h"
+#include <criterion/criterion.h>
+#include <string.h>
+
+static bool compare(const void *elem, const void *data)
```
Comment (aurelenc) [inline tests/tests_llist_search.c:12]: could have returned strcmp directly to get the difference

File: libs/database/database.h (hunk 1 above)
Comment (ZQUEMA) [reply libs/database/database.h]: added

File: tests/tests_llist_search.c [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+Test (none_list, llist_search)
+{
+    llist_t *list = 0;
+
+    cr_assert_eq(0, llist_search(list, &compare, "Hello"));
+}
+
+Test (search_not_find, llist_search)
```
Comment (aurelenc) [inline tests/tests_llist_search.c:38]: not found

File: tests/tests_llist_search.c (hunk 3 above)
Comment (ZQUEMA) [reply tests/tests_llist_search.c:12]: This is a test to make sure it works properly.

File: tests/tests_llist_search.c (hunk 4 above)
Comment (ZQUEMA) [reply tests/tests_llist_search.c]: done

**Review by aurelenc — APPROVED** [review]
LGTM

**Review by TomDUVAL-MAHE — COMMENTED** [review]
Ok

**Pull request merged**